# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/GfxInfo.h
# Commit: ee1cda6fcd58
# Full Hash: ee1cda6fcd58bcf627cb1e52753d5350ebd571f1
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2021-04-28 03:32:08
# Regressor Bug: 1698732
# File Overlap Count: 1
# Description:
#   Bug 1698732 - Remove Win32k usage from GfxInfo in content process r=bas
#   
#   Make it so that GfxInfo::Init() no longer does anything and add some asserts
#   to several GfxInfo APIs to ensure that they are not called by accident from
#   content process, as calling them now will return invalid data.
# ==============================================================================

diff -r f13af8d47dca -r ee1cda6fcd58 widget/windows/GfxInfo.h
--- a/widget/windows/GfxInfo.h	Tue Apr 27 19:37:14 2021 +0000
+++ b/widget/windows/GfxInfo.h	Tue Apr 27 19:45:14 2021 +0000
@@ -5,19 +5,20 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef __mozilla_widget_GfxInfo_h__
-#define __mozilla_widget_GfxInfo_h__
+#ifndef WIDGET_WINDOWS_GFXINFO_H_
+#define WIDGET_WINDOWS_GFXINFO_H_
 
 #include "GfxInfoBase.h"
 
-namespace mozilla {
-namespace widget {
+namespace mozilla::widget {
 
 class GfxInfo : public GfxInfoBase {
-  ~GfxInfo() {}
+ public:
+  using GfxInfoBase::GetFeatureStatus;
+  using GfxInfoBase::GetFeatureSuggestedDriverVersion;
 
- public:
-  GfxInfo();
+  GfxInfo() = default;
+  nsresult Init() override;
 
   // We only declare the subset of nsIGfxInfo that we actually implement. The
   // rest is brought forward from GfxInfoBase.
@@ -55,10 +56,8 @@
   NS_IMETHOD GetDisplayWidth(nsTArray<uint32_t>& aDisplayWidth) override;
   NS_IMETHOD GetDisplayHeight(nsTArray<uint32_t>& aDisplayHeight) override;
   NS_IMETHOD GetDrmRenderDevice(nsACString& aDrmRenderDevice) override;
-  using GfxInfoBase::GetFeatureStatus;
-  using GfxInfoBase::GetFeatureSuggestedDriverVersion;
+  NS_IMETHOD RefreshMonitors() override;
 
-  nsresult Init() override;
   NS_IMETHOD_(int32_t) GetMaxRefreshRate(bool* aMixed) override;
 
   uint32_t OperatingSystemVersion() override { return mWindowsVersion; }
@@ -66,22 +65,11 @@
 
   nsresult FindMonitors(JSContext* cx, JS::HandleObject array) override;
 
-  NS_IMETHOD RefreshMonitors() override;
-
 #ifdef DEBUG
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_NSIGFXINFODEBUG
 #endif
 
- protected:
-  virtual nsresult GetFeatureStatusImpl(
-      int32_t aFeature, int32_t* aStatus, nsAString& aSuggestedDriverVersion,
-      const nsTArray<GfxDriverInfo>& aDriverInfo, nsACString& aFailureId,
-      OperatingSystem* aOS = nullptr) override;
-  virtual const nsTArray<GfxDriverInfo>& GetGfxDriverInfo() override;
-
-  void DescribeFeatures(JSContext* cx, JS::Handle<JSObject*> aOut) override;
-
  private:
   struct DisplayInfo {
     uint32_t mScreenWidth;
@@ -91,7 +79,23 @@
     nsString mDeviceString;
   };
 
- private:
+  ~GfxInfo() = default;
+
+  // Disallow copy/move
+  GfxInfo(const GfxInfo&) = delete;
+  GfxInfo& operator=(const GfxInfo&) = delete;
+  GfxInfo(GfxInfo&&) = delete;
+  GfxInfo& operator=(GfxInfo&&) = delete;
+
+  nsresult GetFeatureStatusImpl(int32_t aFeature, int32_t* aStatus,
+                                nsAString& aSuggestedDriverVersion,
+                                const nsTArray<GfxDriverInfo>& aDriverInfo,
+                                nsACString& aFailureId,
+                                OperatingSystem* aOS = nullptr) override;
+  const nsTArray<GfxDriverInfo>& GetGfxDriverInfo() override;
+
+  void DescribeFeatures(JSContext* cx, JS::Handle<JSObject*> aOut) override;
+
   void AddCrashReportAnnotations();
 
   nsString mDeviceString[2];
@@ -103,16 +107,15 @@
   nsString mAdapterVendorID[2];
   nsString mAdapterDeviceID[2];
   nsString mAdapterSubsysID[2];
-  uint32_t mWindowsVersion;
-  uint32_t mWindowsBuildNumber;
-  uint32_t mActiveGPUIndex;  // This must be 0 or 1
+  uint32_t mWindowsVersion = 0;
+  uint32_t mWindowsBuildNumber = 0;
+  uint32_t mActiveGPUIndex = 0;  // This must be 0 or 1
   nsTArray<DisplayInfo> mDisplayInfo;
-  bool mHasDualGPU;
-  bool mHasDriverVersionMismatch;
-  bool mHasBattery;
+  bool mHasDualGPU = false;
+  bool mHasDriverVersionMismatch = false;
+  bool mHasBattery = false;
 };
 
-}  // namespace widget
-}  // namespace mozilla
+}  // namespace mozilla::widget
 
-#endif /* __mozilla_widget_GfxInfo_h__ */
+#endif  // WIDGET_WINDOWS_GFXINFO_H_
